projects/ion/src/lib/directives/dropdown/dropdown.component.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  computed,
+  model,
+  output,
+  signal,
+} from '@angular/core';
+
+import { IonDropdownOption, IonDropdownProps } from './types';
+import { IonIconComponent } from '../../icon';
+import { IonNoDataComponent } from '../../no-data';
+import { IonSpinnerComponent } from '../../spinner';
+import { IonButtonComponent } from '../../button';
+
+@Component({
+  standalone: true,
+  selector: 'ion-dropdown',
+  templateUrl: 'dropdown.component.html',
+  styleUrls: ['dropdown.component.scss'],
+  imports: [
+    IonIconComponent,
+    IonNoDataComponent,
+    IonSpinnerComponent,
+    IonButtonComponent,
+  ],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class IonDropdownComponent<T extends IonDropdownOption> {
+  dropdownConfig = signal<IonDropdownProps<T>['dropdownConfig']>({});
+  dropdownLoading = signal<IonDropdownProps<T>['dropdownLoading']>(false);
+  dropdownOptions = model<IonDropdownProps<T>['dropdownOptions']>([]);
+  dropdownOptionsChange = output<IonDropdownProps<T>['dropdownOptions']>();
+
+  public selectedOptions = computed(() =>
+    this.dropdownOptions().filter(option => option.selected)
+  );
+
+  public handleHover(hoveredOption: T, mouseEnter: boolean): void {
+    if (hoveredOption.disabled || !hoveredOption.selected) {
+      return;
+    }
+
+    hoveredOption.hovered = mouseEnter;
+  }
+
+  public clearOptions(): void {
+    this.dropdownOptions.update(oldOptions =>
+      oldOptions.map(option => {
+        return option.disabled
+          ? option
+          : {
+              ...option,
+              selected: false,
+            };
+      })
+    );
+  }
+
+  public selectOption(selectedOption: T): void {
+    if (selectedOption.disabled) {
+      return;
+    }
+
+    if (this.dropdownConfig().multiple) {
+      this.handleMultipleOptions(selectedOption);
+      this.dropdownOptionsChange.emit(this.dropdownOptions());
+      return;
+    }
+
+    this.handleSingleOptions(selectedOption);
+    this.dropdownOptionsChange.emit(this.dropdownOptions());
```

Comment:
```suggestion
    if (this.dropdownConfig().multiple) {
      this.handleMultipleOptions(selectedOption);
    } else {
      this.handleSingleOptions(selectedOption);
    }
    
    this.dropdownOptionsChange.emit(this.dropdownOptions());
```

This avoids duplicating `this.dropdownOptionsChange.emit(this.dropdownOptions());`